motoman_driver/src/industrial_robot_client/joint_trajectory_action.cpp

```diff
@@ -62,7 +63,11 @@ JointTrajectoryAction::JointTrajectoryAction() :
   // A parameter named topics_list contains the respective information for each of the groups
 
   std::string value;
-  ros::param::search("topics_list", value);
+  if(!ros::param::search("topics_list", value))
+  {
+    ROS_ERROR_STREAM(industrial_robot_client::motoman_utils::TOPIC_LIST_ERROR_MSG);
+    throw;
```

Comment:
It probably works, but only because (from [here, syntax 2](http://en.cppreference.com/w/cpp/language/throw)):

> This form is only allowed when an exception is presently being handled (it calls [std::terminate](http://en.cppreference.com/w/cpp/error/terminate) if used otherwise).

I'm probably being anal here, but even though the `if` will only be taken when the parameter is not found, the exception that is then thrown (most likely a `std::terminate`) does not convey/encode that error (ie: it's actually thrown because of that `throw` without an active exception, not because of the missing parameter).